### PR TITLE
Email confirmation

### PR DESCRIPTION
**Pull request by jakazzy**
#### What does this PR do?
This PR does the following:
1. sends an email confirmation to the user when the user signs up
2. Resets password when a user forgets password
3. creates a customer error ResourceNotFound to deal with lost resources
4. stores errors with input values(such as email and password in an array) so the error is displayed only after assessing all input values.
5. Moves the **password confirmation** method and **resetpassword** method to the user model
Finally it adds error route to the application

#### Description of Task to be completed?
1. I created ``` confirmed: false``` property on the user model.
2. Using nodemailer, I send an activation link to the user, and the token I use is the user's jwt token

#### How should this be manually tested?
1. Run the server ``` npm start ```
2. Using an API client,  create a new user.
3. After creating the user, an email will be sent to your email address.
4. Click on the activation link in the email
5. This will open a new link in the browser.
6. Login using user details in using an API client.

#### Any background context you want to provide?
There is currently no UI

#### What are the relevant user stories?
This PR is Linked to: https://github.com/jakazzy/exercise-tracker/issues/14

#### Questions:

### CODE REVIEW

**Comment by jakazzy** [comment]
@kofiasare This PR is the latest.
 please don't merge yet.

**Review by kofiasare — COMMENTED** [review]
Extract this functionality into the user Model
so you can have
`model.User.send_**confirmation_instructions**_mail(to, message)
the `\*...*` part is dynamic as you can have
 
e.g `mode.User.send_reset_password_instructions_mail()`

**Comment by jakazzy** [comment]
Sure noted

**Comment by jakazzy** [comment]
@kofiasare pls I have updated PR with updates for both auth and email confirmation

**Comment by jakazzy** [comment]
@kofiasare please kindly review this and ignore the other pull request [auth](https://github.com/jakazzy/exercise-tracker/pull/9)

File: backend/src/controllers/v1/exercisesController.js [hunk 1]
```diff
@@ -1,5 +1,11 @@
 import { initModels as model } from '../../models'
+import { RecordNotFound } from './../../lib/errors'
 
+
+export const setUser = async id => {
```
Comment (kofiasare) [inline backend/src/controllers/v1/exercisesController.js]: Please why are you export this function?

File: backend/src/server.js [hunk 2]
```diff
@@ -40,7 +41,17 @@ import { restrictCors } from './middlewares'
   app.use(bodyParser.json());
   app.use(restrictCors);
   app.use('/api/v1', routers.v1Router(express));
-
+  // route to handle errors
```
Comment (kofiasare) [inline backend/src/server.js:44]: Handle which error?
Does it work?
How does it work?

**Review by kofiasare — CHANGES_REQUESTED** [review]
I think you are overusing the RecordNotFound oo. The idea was that you use it with the setUser function as a way to respond to a situation where a user is not found for the exercise.

File: backend/src/controllers/v1/exercisesController.js (hunk 1 above)
Comment (jakazzy) [reply backend/src/controllers/v1/exercisesController.js]: ok initially I wanted to use it in the user controller as well. I will move it to the exercise controller

File: backend/src/server.js (hunk 2 above)
Comment (jakazzy) [reply backend/src/server.js:44]: so  if a user tries any route eg
{{base_url}}/resource since the path does not exist, the handler is triggered

**Comment by jakazzy** [comment]
@kofiasare currently done with the changes

**Review by kofiasare — APPROVED** [review]

**Pull request merged**